### PR TITLE
[1.20.5] Fix porting todos part 1

### PR DESCRIPTION
**Pull request by Minecraftschurli**
- [ ] `net.minecraft.client.gui.screens.worldselection.WorldOpenFlows.java (68, 46) public class WorldOpenFlows { // TODO 1.20.3 PORTING: re-add the autoconfirm code`
- [ ] `net.minecraft.client.particle.ParticleEngine.java (457, 16) //TODO porting: is this even needed with the particle render order fix???`
- [ ] `net.minecraft.util.datafix.fixes.StructuresBecomeConfiguredFix.java (186, 16) // Porting 1.20.5 check if this is correct`
- [ ] `net.minecraft.world.entity.Entity.java (2803, 217) net.neoforged.neoforge.event.entity.EntityEvent.Size sizeEvent = net.neoforged.neoforge.event.EventHooks.getEntitySizeForge(this, pose, entitydimensions, entitydimensions1, entitydimensions1.eyeHeight()); // Porting 1.20.5 check if this still works`
- [x] `net.minecraft.world.entity.LivingEntity.java (871, 12) // Porting 1.20.5 re-add or remove PotionColorCalculationEvent`
- [ ] `net.minecraft.world.inventory.GrindstoneMenu.java (137, 70) if (this.xp != Integer.MIN_VALUE) return ItemStack.EMPTY; // Porting 1.20.5 check if this is correct`
- [ ] `net.minecraft.world.item.BowItem.java (28, 44) if (!itemstack.isEmpty()) { // Porting 1.20.5 redo EventHooks.onArrowLoose`
- [ ] `net.minecraft.world.level.block.RailBlock.java (168, 57) return p_55402_; // TODO 1.20.3 PORTING: reevaluate`
- [ ] `net.minecraft.world.level.block.StemBlock.java (93, 145) if (p_222539_.isEmptyBlock(blockpos) && (blockstate.is(Blocks.FARMLAND) || blockstate.is(BlockTags.DIRT))) { // TODO 1.20.3 PORTING: reimplement canSustainPlant check`
- [ ] `net.minecraft.world.level.saveddata.maps.MapDecoration.java (12, 119) public record MapDecoration(Holder<MapDecorationType> type, byte x, byte y, byte rot, Optional<Component> name) {  // Porting 1.20.5 this is synced now reevaluate shouldRenderForIndex`
- [ ] `net.neoforged.neoforge.client.gui.widget.ModListWidget.java (37, 47) //this.setRenderBackground(false); // Porting 1.20.5 still needed?`
- [ ] `net.neoforged.neoforge.common.BasicItemListing.java (55, 120) ItemCost cost = new ItemCost(price.getItemHolder(), price.getCount(), DataComponentPredicate.EMPTY, price); // Porting 1.20.5 do something proper for the components here`
- [x] `net.neoforged.neoforge.common.CommonHooks.java (521, 12) // Porting 1.20.5 redo this for components?`
- [x] `net.neoforged.neoforge.debug.attachment.AttachmentTests.java (30, 35) public class AttachmentTests { // Porting 1.20.5 nuke?`
- [x] `net.neoforged.neoforge.debug.item.ItemTests.java (133, 16) // Porting 1.20.5 replacement?`
- [ ] `net.neoforged.neoforge.debug.loot.GlobalLootModifiersTest.java (142, 100) return loottable.getRandomItems(builder.create(LootContextParamSets.EMPTY)); // TODO - porting: we need an AT`
- [x] `net.neoforged.neoforge.oldtest.item.HiddenTooltipPartsTest.java (55, 17) // TODO PORTING 1.20.5 - fix or remove test`
- [x] remove stuff annotated with `@Deprecated(forRemoval=true)` with a since version less than 1.20.5

### CODE REVIEW

**Comment by Spinoscythe** [comment]
@Minecraftschurli Please could you remove the AT for `MenuScreens.register` since `RegisterMenuScreensEvent` helps with that now?

**Comment by pupnewfster** [comment]
In relation to the todo about the size event. Does it even serve a purpose anymore or can we just replace usages of it with using the vanilla size attribute?

**Comment by Shadows-of-Fire** [comment]
The size event allows changing the width separately from the height (which scale does not allow).  I'm not convinced that usecase is worth keeping it though.

File: tests/src/main/java/net/neoforged/neoforge/oldtest/item/HiddenTooltipPartsTest.java [hunk 1]
```diff
@@ -1,58 +0,0 @@
...
-import net.minecraft.world.item.component.ItemAttributeModifiers;
-import net.neoforged.bus.api.IEventBus;
-import net.neoforged.fml.common.Mod;
-import net.neoforged.neoforge.event.BuildCreativeModeTabContentsEvent;
-import net.neoforged.neoforge.registries.DeferredItem;
-import net.neoforged.neoforge.registries.DeferredRegister;
-
-@Mod(HiddenTooltipPartsTest.MOD_ID)
-public class HiddenTooltipPartsTest {
-    public static final String MOD_ID = "hidden_tooltip_parts";
```
Comment (Technici4n) [inline tests/src/main/java/net/neoforged/neoforge/oldtest/item/HiddenTooltipPartsTest.java:24]: Shouldn't this be removed from `mods.toml`?

**Review by Technici4n — CHANGES_REQUESTED** [review]

**Review by Technici4n — DISMISSED** [review]

**Comment by Technici4n** [comment]
You still have merge conflicts.

**Review by Technici4n — APPROVED** [review]

**Pull request merged**